### PR TITLE
[JBPM-10001] Variable name might be null in IntermediateThrowHandler

### PR DESCRIPTION
**Pull request by fjtirado**
**JIRA**: 

[link](https://issues.redhat.com/browse/JBPM-10001)

### CODE REVIEW

File: jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/IntermediateThrowEventHandler.java [hunk 1]
```diff
@@ -241,9 +241,15 @@ public void handleMessageNode(final Node node, final Element element,
 							"Could not find message " + messageRef);
 				}
                 message.addOutgoingNode(node);
+                String varName = null;
                 String mappingVariable = (String) actionNode.getMetaData("MappingVariable");
                 Variable v = (Variable) ((ProcessBuildData) parser.getData()).getMetaData("Variable");
-                String varName = v != null ? (String) v.getMetaData(mappingVariable) : mappingVariable;
+                if (v != null) {
```
Comment (fjtirado) [inline jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/IntermediateThrowEventHandler.java:247]: In previous implementation if Variable was not null but Variable.getMedata(mappingVariable) was, then rather that use mappingVariable (as I think should be the case) we will be setting varName to null. 
This makes the message object used by the action (which is also used for Kafka  extension) null.

File: jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/IntermediateThrowEventHandler.java [hunk 2]
```diff
@@ -325,6 +331,7 @@ protected void readDataInputAssociation(org.w3c.dom.Node xmlNode,
             subNode = subNode.getNextSibling();
             if (subNode != null && "transformation".equals(subNode.getNodeName())) {
                 String lang = subNode.getAttributes().getNamedItem("language").getNodeValue();
+                if (text.startsWith("\"") && text.endsWith("\"")) {
```
Comment (gmunozfe) [inline jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/IntermediateThrowEventHandler.java:334]: Compilation error, missing closing bracket:
[ERROR] /home/jenkins/workspace/KIE/main/pullrequest/jbpm-main.pr/bc/kiegroup_jbpm/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/IntermediateThrowEventHandler.java:[382,9] illegal start of expression

**Review by gmunozfe — CHANGES_REQUESTED** [review]

File: jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/IntermediateThrowEventHandler.java [hunk 3]
```diff
@@ -241,9 +240,7 @@ public void handleMessageNode(final Node node, final Element element,
 							"Could not find message " + messageRef);
 				}
                 message.addOutgoingNode(node);
-                String mappingVariable = (String) actionNode.getMetaData("MappingVariable");
-                Variable v = (Variable) ((ProcessBuildData) parser.getData()).getMetaData("Variable");
-                String varName = v != null ? (String) v.getMetaData(mappingVariable) : mappingVariable;
+                String varName = (String) actionNode.getMetaData("MappingVariable");
```
Comment (fjtirado) [inline jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/IntermediateThrowEventHandler.java:243]: The var name should be retireved in the same way than in EndEventHandler

**Review by gmunozfe — APPROVED** [review]
Looks good to me, well done @fjtirado !

**Pull request merged**